data_structures/hash_table/test_hash_table.py

```diff
@@ -0,0 +1,104 @@
+from hash_table import HashTable
+import pytest
+
+
+def test_TypeError_hash_key(empty_HT):
+    """tests that TypeError is raised when non-string key used."""
+    with pytest.raises(TypeError):
+        empty_HT.hash_key(1)
+    with pytest.raises(TypeError):
+        empty_HT.hash_key([1, 2, 3])
+    with pytest.raises(TypeError):
+        empty_HT.hash_key({'test': 8})
```

Comment:
Each of these should be its own test since they are checking if various types cause a `TypeError`.